expr/func_linearregression_test.go

```diff
@@ -0,0 +1,125 @@
+package expr
+
+import (
+	"testing"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/schema"
+)
+
+func TestLinearRegressionInvalidStartSourceAt(t *testing.T) {
+	funcLinearRegression := FuncLinearRegression{
+		startSourceAt: "test",
+	}
+
+	_, err := funcLinearRegression.Exec(initDataMap([]models.Series{}))
+	if err == nil {
+		t.Fatal("invalid 'startSourceAt' should result in error")
+	}
+}
+
+func TestLinearRegressionInvalidEndSourceAt(t *testing.T) {
+	funcLinearRegression := FuncLinearRegression{
+		endSourceAt: "test",
+	}
+
+	_, err := funcLinearRegression.Exec(initDataMap([]models.Series{}))
+	if err == nil {
+		t.Fatal("invalid 'endSourceAt' should result in error")
+	}
+}
+
+func TestLinearRegression(t *testing.T) {
+	in := []models.Series{{
+		Target: "test.value",
+		Tags: map[string]string{
+			"test": "value",
+		},
+		QueryPatt: "test.value",
+		Interval:  60,
+		QueryFrom: 120,
+		QueryTo:   540,
+		Datapoints: []schema.Point{
+			{
```

Comment:
Should include some test with NaNs

Reply:
This test actually does include "NaN"s (although here they're represented as missing data points, which I'll update)